modules/tests/shared/src/test/scala/lucuma/core/math/SignalToNoiseSuite.scala

```diff
@@ -15,24 +18,30 @@ import lucuma.core.optics.Format
 import lucuma.core.optics.laws.discipline.FormatTests
 import lucuma.core.optics.laws.discipline.ValidSplitEpiTests
 import monocle.law.discipline.*
+import org.scalacheck.Arbitrary
+import org.scalacheck.Gen
 import org.scalacheck.Prop.*
 
 
 final class SignalToNoiseSuite extends munit.DisciplineSuite {
 
   import ArbSignalToNoise.given
   import ArbSignalToNoise.bigDecimalSignalToNoise
-  import ArbSignalToNoise.posBigDecimalSignalToNoise
+  import ArbSignalToNoise.nonNegBigDecimalSignalToNoise
   import ArbSignalToNoise.stringSignalToNoise
 
+  given Arbitrary[NonNegBigDecimal] = Arbitrary {
+    Gen.posNum[BigDecimal].map(refineV[NonNegative](_).getOrElse(throw new RuntimeException))
+  }
+
   checkAll("Order",                     OrderTests[SignalToNoise].order)
   checkAll("JSON Codec",                CodecTests[SignalToNoise].codec)
   checkAll("FromBigDecimalExact",       PrismTests(SignalToNoise.FromBigDecimalExact))
   checkAll("FromBigDecimalRounding",
     ValidSplitEpiTests(SignalToNoise.FromBigDecimalRounding).validSplitEpiWith(bigDecimalSignalToNoise))
-  checkAll("FromPosBigDecimalExact",    PrismTests(SignalToNoise.FromPosBigDecimalExact))
+  checkAll("FromPosBigDecimalExact",    PrismTests(SignalToNoise.FromNonNegBigDecimalExact))
```

Comment:
I suppose we could update the label as well since it still reads "FromPosBigDecimal".